src/app/parcels/add/formSections/CollectionDateCard.tsx

```diff
@@ -0,0 +1,28 @@
+import React from "react";
+import { CardProps, onChangeDate, errorText } from "@/components/Form/formFunctions";
+import GenericFormCard from "@/components/Form/GenericFormCard";
+import StyledDatePicker from "@/components/DataInput/DatePicker";
+import { ErrorText } from "@/components/Form/formStyling";
+
+const CollectionDateCard: React.FC<CardProps> = ({ fieldSetter, errorSetter, formErrors }) => {
+    return (
+        <GenericFormCard
+            title="Collection Date Card"
+            required={false}
+            text="What date is the client collecting their parcel?"
+        >
+            <>
+                <StyledDatePicker
+                    onChange={(value: any): void => {
+                        const newValue = value as Date | null;
+                        onChangeDate(fieldSetter, errorSetter, "collectionDate", newValue);
+                    }}
+                    label="Enter Date Here"
+                />
+                <ErrorText>{errorText(formErrors.collectionDate)}</ErrorText>
```

Comment:
I think MUI has something similar here

https://mui.com/x/react-date-pickers/validation/#show-the-error